src/optimize/dynamic_dll_plugin/dynamic_dll_plugin.js

```diff
@@ -456,12 +291,43 @@ export class DynamicDllPlugin {
     return stubModule;
   }
 
+  async areMaxCompilationsPerformed() {
+    // Logic to run the max compilation requirements.
+    // Only enable this for CI builds in order to ensure
+    // we have an healthy dll ecosystem.
+    if (this.performedCompilations === this.maxCompilations) {
+      return Promise.reject('All the allowed dll compilations were already performed and one more is needed which is not possible');
+    }
+
+    return Promise.resolve();
```

Comment:
This is unnecessary as the async function will always resolve if it completes without throwing

Reply:
👍